mhw_analysis/ai/ai_mhw_images.py

```diff
@@ -0,0 +1,295 @@
+  """
+  Queries MHW_Events table given ranges (latitude, longitude, date), and returns
+  a dataframe containing all the entries of the table which meet that criteria. 
+
+  Query (SQL):
+  SELECT * FROM MHW_Events 
+  WHERE lat BETWEEN ? AND ? 
+  AND lon BETWEEN ? AND ?
+  AND date BETWEEN ? AND ?
+
+  :param table:
+    SQLAlchemy Table object containing MHW_Events
+    Run the following code before calling this function:
+      engine = sqlalchemy.create_engine('sqlite:///'+mhw_file)
+      metadata = sqlalchemy.MetaData()
+      mhw_tbl = sqlalchemy.Table('MHW_Events', metadata, autoload=True, autoload_with=engine)
+  
+  :param lat_start:
+    double representing the lower end of latitude range
+  
+  :param lat_end:
+    double representing the upper end of latitude range
+  
+  :param lon_start:
+    double representing the lower end of longitude range
+  
+  :param lon_end:
+    double representing the lower end of longitude range
+  
+  :param date_start:
+    String representing the start date
+    Format: 'Mon Day Year' where Day and Year are integers
+  
+  :param date_end:
+    String representing the start date
+    Format: 'Mon Day Year' where Day and Year are integers
+  
+  :return range_df:
+    Pandas dataframe containing query results
+  """
+
+  # can't run if table not properly set up
+  # if not table:
+  #   print("Please load the MHW_Events table into a Table object.")
+  #   return None
+  
+  # set up query
+  query = sqlalchemy.select([table]).where(sqlalchemy.and_(
+    # range of lat 
+    sqlalchemy.between(
+        # pass functions?
+        table.columns.lat, lat_start, lat_end
+    ),
+    # and range of lon
+    sqlalchemy.between(
+        # pass functions?
+        table.columns.lon, lon_start, lon_end
+    ),
+    # and range of dates
+    sqlalchemy.between(
+        # pass functions?
+        table.columns.date, datetime.strptime(date_start, '%b %d %Y'), datetime.strptime(date_end, '%b %d %Y')
+    )
+    ))
+  
+  # execute query
+  result = connection.execute(query)
+
+  # get df
+  range_df = DataFrame(result.fetchall())
+  # range_df.columns = query.keys()
+
+  # return df
+  return range_df
+
+# used to save the tuples 
+# could include folder path as a param
+def save_obj(obj, name):
+	# save object to folder
+    with open('mhw_data/'+ name + '.pkl', 'wb') as f:
+        pickle.dump(obj, f, pickle.HIGHEST_PROTOCOL)
+
+def load_obj(name):
+    # load object from folder
+    with open('mhw_data/' + name + '.pkl', 'rb') as f:
+        return pickle.load(f)
+
+def get_mhw_tbl(mhw_file):
+	# create engine and check if it has the table we need
+	engine = sqlalchemy.create_engine('sqlite:///'+mhw_file)
+	engine.has_table("MHW_Events")
+
+	# connect to engine
+	connection = engine.connect()
+
+	# load table from metadata
+	metadata = sqlalchemy.MetaData()
+	mhw_tbl = sqlalchemy.Table('MHW_Events', metadata, autoload=True, autoload_with=engine)
```

Comment:
Does this load the whole (10+Gb) table into memory?
If so, that must be very slow.  But maybe this is just
staging the table.  Shows my SQL ignorance.

Reply:
Just staging the table, my computer would be dying if it loaded the whole thing into memory